dagql/idproto/id.proto

```diff
@@ -0,0 +1,98 @@
+syntax = "proto3";
+package dagger;
+
+option go_package = "./idproto";
+
+// ID represents a GraphQL value of a certain type, constructed by evaluating
+// its contained pipeline. In other words, it represents a
+// constructor-addressed value, which may be an object, an array, or a scalar
+// value.
+//
+// It may be binary=>base64-encoded to be used as a GraphQL ID value for
+// objects. Alternatively it may be stored in a database and referred to via an
+// RFC-6920 ni://sha-256;... URI.
+message ID {
```

Comment:
I think we'll want to change this pretty soon to be more compact. It's really easy to have arguments or parents that depend on the same ID by carrying it around wholesale, inflating the ID payload through repetition. This is especially common now that IDs contain a reference to their originating module.

We should probably restructure this using a flat `map[digest.Digest]ID` and have embedded IDs reference digests instead.